### PR TITLE
Feature/co 1587 implement unit testing

### PR DESCRIPTION
**Pull request by kalcicd**

### CODE REVIEW

File: tests/unit/owners-serializer.test.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    const serializedOwners = serializeOwners(rawOwners);
+    testMultipleResources(serializedOwners, rawOwners, resourceType, 'id');
+    _.forEach(serializedOwners.data, (value) => {
+      expect(value).to.have.all.keys(_.keys(ownerResourceDef));
+      expect(value.attributes).to.have.all.keys(_.keys(ownerResourceDef.attributes.properties));
+    });
+  });
+
+  /*
+  it('serializeOwner should be rejected if given null arguments', () => {
```
Comment (kalcicd) [inline tests/unit/owners-serializer.test.js]: When serializeOwner is given a null argument, the serializer throws an error because rawOwner.id is called to generate the top level self link. However, when null is passed into serializeOwners, it just returns an object with a null data key. What is the correct response here? Or do I not need to worry about this edge case to begin with?

Comment (ianbrown9475) [reply tests/unit/owners-serializer.test.js]: I'm thinking we won't need to worry about this one. When you query the database, I think the value of `rows` will always be an array of non-null values.

File: tests/unit/parks-dao.test.js [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+    parksDao = proxyquire(`${appRoot}/api/v1/db/oracledb/parks-dao`, {
+      '../../serializers/parks-serializer': {
+        serializePark: serializeParkStub,
+        serializeParks: serializeParksStub,
+      },
+    });
+  });
+  afterEach(() => sinon.restore());
+
+  describe('Test getParks', () => {
```
Comment (kalcicd) [inline tests/unit/parks-dao.test.js]: Is there any need to test queries for getParks? I don't believe its possible to get any invalid query params because of the openapi constraints.

Comment (ianbrown9475) [reply tests/unit/parks-dao.test.js]: It depends on how much you want to test. You could pass in different query parameters and check that the sql query it uses is correct. Up to you.

File: tests/unit/test-util.js [hunk 3]
```diff
@@ -86,15 +97,23 @@ const testSingleResource = (serializedResource, resourceType, resourceId, nested
...
+const testMultipleResources = (serializedResources,
+  rawResources, resourceType, resourceKey, relationships) => {
   expect(serializedResources).to.have.all.keys('data', 'links');
   expect(serializedResources.data).to.be.an('array');
-  _.zipWith(serializedResources.data, rawResources, (a, b) => {
-    expect(a).to.deep.equal(resourceSchema(resourceType,
-      b[resourceKey],
-      _.omit(b, resourceKey)).data);
-  });
+  for (let i = 0; i < serializedResources.data.length; i += 1) {
```
Comment (ianbrown9475) [inline tests/unit/test-util.js]: You should be able to use [`_.zipWith`](https://lodash.com/docs/4.17.15#zipWith) here

File: tests/unit/test-util.js [hunk 4]
```diff
@@ -86,15 +97,23 @@ const testSingleResource = (serializedResource, resourceType, resourceId, nested
  * @param {object} rawResources Raw resources to be used in test.
  * @param {*} resourceType The type of the resource.
  * @param {*} resourceKey The id of the resource.
+ * @param {object} relationships The relationships of the resource.
  */
-const testMultipleResources = (serializedResources, rawResources, resourceType, resourceKey) => {
+const testMultipleResources = (serializedResources,
+  rawResources, resourceType, resourceKey, relationships) => {
```
Comment (ianbrown9475) [inline tests/unit/test-util.js]: Could you fix the indentation here? https://github.com/airbnb/javascript#functions--signature-invocation-indentation

File: tests/unit/test-util.js [hunk 5]
```diff
@@ -72,11 +80,14 @@ const getDefinition = def => openapi.definitions[def].properties;
...
+ * @param {[object]} relationships The relationships of the resource.
  */
-const testSingleResource = (serializedResource, resourceType, resourceId, nestedProps) => {
+const testSingleResource = (serializedResource,
+  resourceType, resourceId, nestedProps, relationships) => {
   expect(serializedResource).to.deep.equal(resourceSchema(resourceType,
     resourceId,
-    nestedProps));
+    nestedProps,
+    relationships));
```
Comment (ianbrown9475) [inline tests/unit/test-util.js]: Similar comment on indentation https://github.com/airbnb/javascript#functions--signature-invocation-indentation
